src/components/tooltip/CallNode.js

```diff
@@ -23,23 +35,134 @@ type Props = {|
   // label, e.g. "100ms" or "33%".
   durationText: string,
   callTree?: CallTree,
+  timings?: TimingsForPath,
 |};
 
+function _getFriendlyStackTypeName(
+  implementation: StackImplementation
+): string {
+  switch (implementation) {
+    case 'ion':
+    case 'baseline':
+      return `JS JIT (${implementation})`;
+    case 'interpreter':
+      return 'JS interpreter';
+    case 'native':
+      return 'Native code';
+    case 'unknown':
+      return implementation;
+    default:
+      throw assertExhaustiveCheck(implementation);
+  }
+}
+
 /**
  * This class collects the tooltip rendering for anything that cares about call nodes.
  * This includes the Flame Graph and Stack Chart.
  */
 export class TooltipCallNode extends React.PureComponent<Props> {
+  _renderTimings(
+    maybeTimings: ?TimingsForPath,
+    maybeDisplayData: ?CallNodeDisplayData
+  ) {
+    if (!maybeTimings || !maybeDisplayData) {
+      return null;
+    }
+    const timings = maybeTimings;
+    const displayData = maybeDisplayData;
+    const { breakdownByImplementation } = timings.forPath.totalTime;
+    if (!breakdownByImplementation) {
+      return null;
+    }
+    const sortedBreakdown = objectEntriesMap(breakdownByImplementation).sort(
+      (a, b) => b[1] - a[1]
+    );
+
+    const totalTime = sortedBreakdown.reduce(
+      (memo, [, value]) => memo + value,
+      0
+    );
+
+    return (
+      <div className="tooltipCallNodeImplementation">
+        {/* grid row -------------------------------------------------- */}
+        <div />
+        <div className="tooltipCallNodeImplementationHeader" />
+        <div className="tooltipCallNodeImplementationHeader">
+          <span className="tooltipCallNodeImplementationHeaderSwatchRunning" />Running
+        </div>
+        <div className="tooltipCallNodeImplementationHeader">
+          <span className="tooltipCallNodeImplementationHeaderSwatchSelf" />Self
+        </div>
+        {/* grid row -------------------------------------------------- */}
+        <div className="tooltipLabel">Overall</div>
+        <div className="tooltipCallNodeImplementationGraph">
+          <div
+            className="tooltipCallNodeImplementationGraphRunning"
+            style={{
+              width: GRAPH_WIDTH,
+            }}
+          />
+          <div
+            className="tooltipCallNodeImplementationGraphSelf"
+            style={{
+              width: GRAPH_WIDTH * displayData.selfTimeNumeric / totalTime,
+            }}
+          />
+        </div>
+        <div>{displayData.totalTimeWithUnit}</div>
+        <div>{displayData.selfTimeWithUnit}</div>
+        {/* grid row -------------------------------------------------- */}
+        {sortedBreakdown.map(([implementation, time], index) => {
+          let selfTime = 0;
+          const selfBreakdownMap =
+            timings.forPath.selfTime.breakdownByImplementation;
```

Comment:
optional nit: you can define this outside of the loop, like close to `timings.forPath.totalTime` above. Maybe something like:
```js
const { totalTime: totalTimings, selfTime: selfTimings } = timings.forPath;
```
Then use these:
```
// totalTimings.value / totalTimings.breakdownByImplementation
// selfTimings.value / selfTimings.breakdownByImplementation
```